internal/scheduler/legacyscheduler.go

```diff
@@ -200,110 +226,166 @@ func (it *QueueCandidateJobsIterator) schedulingReportFromJob(ctx context.Contex
-	for resourceType, quantity := range jobTotalResourceRequests {
-		q := totalQueueResources.Resources[resourceType]
-		q.Add(quantity)
-		totalQueueResources.Resources[resourceType] = q
-
-		q = roundQueueResources.Resources[resourceType]
-		q.Add(quantity)
-		roundQueueResources.Resources[resourceType] = q
-
-		rl := totalQueueResourcesByPriority[jobPriority]
-		if rl.Resources == nil {
-			rl.Resources = make(map[string]resource.Quantity)
-		}
-		q = rl.Resources[resourceType]
-		q.Add(quantity)
-		rl.Resources[resourceType] = q
-		totalQueueResourcesByPriority[jobPriority] = rl
+	// Priority and total requests for this job.
+	priority, _ := PriorityFromJob(job, it.PriorityClasses)
+	jobTotalResourceRequests := schedulerobjects.ResourceList{
+		Resources: common.TotalJobResourceRequest(job),
 	}
 
 	// Check that the job is large enough for this executor.
-	if ok, reason := it.jobIsLargeEnough(jobTotalResourceRequests); !ok {
+	if ok, reason := jobIsLargeEnough(jobTotalResourceRequests, it.MinimumJobSize); !ok {
 		jobSchedulingReport.UnschedulableReason = reason
 		return jobSchedulingReport, nil
 	}
 
-	// Check total per-queue resource limits.
-	if exceeded, reason := it.exceedsResourceLimits(
+	// MaximalResourceFractionToSchedulePerQueue check.
+	roundQueueResourcesByPriority := it.QueueSchedulingRoundReport.ScheduledResourcesByPriority.DeepCopy()
+	roundQueueResourcesByPriority.AddResouceList(priority, jobTotalResourceRequests)
+	if exceeded, reason := exceedsResourceLimits(
 		ctx,
-		totalQueueResources,
-		it.SchedulingConfig.MaximalResourceFractionPerQueue,
+		roundQueueResourcesByPriority.AggregateByResource(),
+		it.SchedulingConstraints.TotalResources,
+		it.MaximalResourceFractionToSchedulePerQueue,
 	); exceeded {
-		jobSchedulingReport.UnschedulableReason = reason + " (total limit for this queue)"
+		jobSchedulingReport.UnschedulableReason = reason + " (per scheduling round limit for this queue)"
 		return jobSchedulingReport, nil
 	}
 
-	// check total per-queue, per priority resource limit
-	if exceeded, reason := it.exceedsPerPriorityResourceLimits(
+	// MaximalResourceFractionPerQueue check.
+	totalQueueResourcesByPriority := it.QueueSchedulingRoundReport.InitialResourcesByPriority.DeepCopy()
+	totalQueueResourcesByPriority.Add(roundQueueResourcesByPriority)
+	if exceeded, reason := exceedsResourceLimits(
 		ctx,
-		jobPriority,
-		totalQueueResourcesByPriority,
-		it.LegacyScheduler.CumulativePriorityLimits,
+		totalQueueResourcesByPriority.AggregateByResource(),
+		it.SchedulingConstraints.TotalResources,
+		it.MaximalResourceFractionPerQueue,
 	); exceeded {
 		jobSchedulingReport.UnschedulableReason = reason + " (total limit for this queue)"
 		return jobSchedulingReport, nil
 	}
 
-	// Check per-round resource limits for this queue.
-	if exceeded, reason := it.exceedsResourceLimits(
+	// MaximalCumulativeResourceFractionPerQueueAndPriority check.
+	if exceeded, reason := exceedsPerPriorityResourceLimits(
```

Comment:
as discussed- this needs to be checked for the whole gang.

Reply:
Done